dev/tools/Magento/Tools/Di/compiler.php

```diff
@@ -33,12 +33,14 @@
             'extra-classes-file=s' => 'path to file with extra proxies and factories to generate',
             'generation=s'         => 'absolute path to generated classes, <magento_root>/var/generation by default',
             'di=s'                 => 'absolute path to DI definitions directory, <magento_root>/var/di by default',
+            'exclude-pattern=s'    => 'allows to exclude Paths from compilation (default is /\/m1\//)',
         ]
     );
     $opt->parse();
 
     $generationDir = $opt->getOption('generation') ? $opt->getOption('generation') : $rootDir . '/var/generation';
     $diDir = $opt->getOption('di') ? $opt->getOption('di') : $rootDir . '/var/di';
+    $fileExcludePatterns = $opt->getOption('exclude-pattern') ? [$opt->getOption('exclude-pattern')] : ['/\/m1\//'];
```

Comment:
@Flyingmana 
Is "m1" as default exclude pattern Ok? 
Does it make sense event specify default? Maybe better to keep it empty?
Thoughts?